src/java.base/share/classes/java/lang/reflect/code/bytecode/BytecodeGenerator.java

```diff
@@ -341,7 +341,28 @@ private void processOperand(Value operand) {
     }
 
     private void processOperands(Op op) {
-        processOperands(op.operands());
+        if (op instanceof InvokeOp invokeOp && invokeOp.isVarArgs()) {
+            processOperands(invokeOp.operands().subList(0, invokeOp.operands().size() - invokeOp.varArgOperands().size()));
+
+            var varArgOperands = invokeOp.varArgOperands();
+            cob.loadConstant(varArgOperands.size());
+            var compType = ((ArrayType) invokeOp.invokeDescriptor().type().parameterTypes().getLast()).componentType();
+            var typeKind = TypeKind.fromDescriptor(compType.toNominalDescriptor().descriptorString());
+            if (TypeKind.REFERENCE.equals(typeKind)) {
+                var cd = ClassDesc.ofDescriptor(compType.toNominalDescriptor().descriptorString());
+                cob.anewarray(cd);
+            } else {
+                cob.newarray(typeKind);
+            }
+            for (int i = 0; i < varArgOperands.size(); i++) {
+                cob.dup();
+                cob.loadConstant(i);
+                load(varArgOperands.get(i));
+                cob.arrayStore(typeKind);
+            }
+        } else {
+            processOperands(op.operands());
+        }
```

Comment:
There are 19 calls of this helper method and only one is related to InvokeOp operands.
I would suggest to apply this logic where InvokeOp bytecode is generated.

Reply:
Yes, keep it local to the `InvokeOp` case at line 883.

Reply:
Add a test using double or long.